flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -102,4 +105,154 @@ public void testLimitPushDown() {
     Assert.assertEquals("should have 1 record", 1, mixedResult.size());
     Assert.assertArrayEquals("Should produce the expected records", mixedResult.get(0), new Object[] {1, "a"});
   }
+
+  @Test
+  public void testFilterPushDown() {
```

Comment:
Tests should be broken into individual methods that are each a test case. To share code, use `@Before` and `@After` and different test suites.

Reply:
Can you also add a test case that listens for a `ScanEvent` and validates that the expression was correctly passed to Iceberg?

Reply:
I add a listener to validate the pushdown for filter in [TestFlinkTableSource](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java)